modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/passthru/PassThroughHttpSender.java

```diff
@@ -336,60 +335,77 @@ public void removePreserveHttpHeader(String headerName) {
         targetConfiguration.getPreserveHttpHeaders().remove(headerName.toUpperCase());
     }
 
-	private void sendRequestContent(final MessageContext msgContext) throws AxisFault {
-
-        if (Boolean.TRUE.equals(msgContext.getProperty(PassThroughConstants.MESSAGE_BUILDER_INVOKED))) {
-            synchronized (msgContext) {
-                while (!Boolean.TRUE.equals(msgContext.getProperty(PassThroughConstants.WAIT_BUILDER_IN_STREAM_COMPLETE)) &&
-                        !Boolean.TRUE.equals(msgContext.getProperty("PASSTHRU_CONNECT_ERROR"))) {
-                    try {
-                        msgContext.wait();
-                    } catch (InterruptedException e) {
-                        e.printStackTrace();
-                    }
+    private boolean waitForReady(final MessageContext msgContext) {
+        synchronized (msgContext) {
+            while (!Boolean.TRUE.equals(msgContext.getProperty(PassThroughConstants.WAIT_BUILDER_IN_STREAM_COMPLETE)) &&
+                   !Boolean.TRUE.equals(msgContext.getProperty("PASSTHRU_CONNECT_ERROR"))) {
+                try {
+                    msgContext.wait();
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
```

Comment:
please avoid using printStackTrace(). Let's log the error stack trace with a message using log.error()

Reply:
fixed